src/seismometer/table/analytics_table.py

```diff
@@ -0,0 +1,715 @@
+                if not self.score_columns or not self.target_columns:
+                    raise ValueError(
+                        "When df is provided, both 'score_columns' and 'target_columns' need to be provided as well."
+                    )
+
+    @property
+    def df(self):
+        return self._df
+
+    @df.setter
+    def df(self, value):
+        self._df = value
+        self._validate_df_statistics_data()
+
+    @property
+    def statistics_data(self):
+        return self._statistics_data
+
+    @statistics_data.setter
+    def statistics_data(self, value):
+        self._statistics_data = value
+        self._validate_df_statistics_data()
+
+    @property
+    def score_columns(self):
+        return self._score_columns
+
+    @score_columns.setter
+    def score_columns(self, value):
+        self._score_columns = value
+        self._validate_df_statistics_data()
+
+    @property
+    def target_columns(self):
+        return self._target_columns
+
+    @target_columns.setter
+    def target_columns(self, value):
+        self._target_columns = value
+        self._validate_df_statistics_data()
+
+    @property
+    def metric(self):
+        return self._metric
+
+    @metric.setter
+    def metric(self, value):
+        try:
+            self._metric = Metric(value.lower()).name
+        except ValueError as e:
+            raise ValueError(
+                f"Invalid metric name: {value}. The metric needs to be one of: "
+                f"{[member.value for member in Metric]}"
+            ) from e
+
+    @property
+    def metrics_to_display(self):
+        return self._metrcis_to_display
+
+    @metrics_to_display.setter
+    def metrics_to_display(self, value):
+        for metric in value:
+            if metric.lower() not in GENERATED_COLUMNS and metric not in [THRESHOLD] + STATNAMES + OVERALL_PERFORMANCE:
+                raise ValueError(
+                    f"Invalid metric name: {value}. The metric needs to be one of: {GENERATED_COLUMNS.keys()}"
+                )
+        self._metrcis_to_display = value if value else list(GENERATED_COLUMNS.keys())
+
+    @property
+    def metric_values(self):
+        return self._metric_values
+
+    @metric_values.setter
+    def metric_values(self, value):
+        self._metric_values = sorted([round(num, self.decimals) for num in value]) if self.df is not None else []
+        self._metric_values = [0 if val == 0.0 else val for val in self._metric_values]
+
+    @property
+    def top_level(self):
+        return self._top_level
+
+    @top_level.setter
+    def top_level(self, value):
+        try:
+            self._top_level = TopLevel(value.lower()).name
+        except ValueError as e:
+            raise ValueError(
+                f"Invalid top_level name: {value}. The top_level needs to be one of: "
+                f"{[member.value for member in TopLevel]}"
+            ) from e
+
+    def generate_initial_table(self, data):
```

Comment:
Can we make PerformanceMetrics be only about computing metrics or only about displaying the table and not both?

BinaryPerformanceMetricsTable could take in an array of metrics (by target, scores, etc) and and then it just needs to format them. 

Then we have some separation from the display layer (great tables) and the content (metrics by target/score) which should make it easier to adopt a new display format in the future. (I didn't do a good job of this in fairness.py, which led to fairness_table being an onerous function)

Reply:
There are two calculations, one is just adding a copy of a column that we want to show as color bar (by default columns_show_bar = None), formatting calculations from binary_performance (for each row,target pair send it to binary_performance) and aligning+joining the dataframe with statistics_data if it is provided. 
So the suggestion is to effectively move _generate_table_data() and _prepare_data(data) out?